libp2p/muxers/muxer.nim

```diff
@@ -45,15 +45,21 @@ proc newMuxerProvider*(creator: MuxerConstructor, codec: string): MuxerProvider
 
 method init(c: MuxerProvider) =
   proc handler(conn: Connection, proto: string) {.async, gcsafe, closure.} =
-    let muxer = c.newMuxer(conn)
-    var handlerFut = if not isNil(c.muxerHandler):
-      c.muxerHandler(muxer)
-    else:
-      var dummyFut = newFuture[void]()
-      dummyFut.complete(); dummyFut
+    let
+      muxer = c.newMuxer(conn)
 
     if not isNil(c.streamHandler):
       muxer.streamHandler = c.streamHandler
 
-    await allFutures(muxer.handle(), handlerFut)
+    var futs = newSeq[Future[void]]()
+
+    futs &= muxer.handle()
+
+    # finally await both the futures
+    if not isNil(c.muxerHandler):
+      futs &= c.muxerHandler(muxer)
+
+    # TODO FIXME what to do with erros here??
```

Comment:
@dryajov Should we just leave `allFutures` and let it throw here?

Reply:
yeah, my original assumption with `allFutures` was that it would throw on errors, so all `allFutures` used are assumed to throw.